client/my-sites/current-site/index.jsx

```diff
@@ -79,11 +80,7 @@ const CurrentSite = React.createClass( {
 	},
 
 	getSelectedSite: function() {
-		if ( this.props.sites.get().length === 1 ) {
-			return this.props.sites.getPrimary();
-		}
-
-		return this.props.sites.getSelectedSite();
+		return this.props.selectedSite || this.props.sites.getPrimary();
```

Comment:
I think that's fair. I suggested to @brunoscopelliti that we could port over the `getPrimary` as a selector in [here](https://github.com/Automattic/wp-calypso/pull/9624#issuecomment-271423871). When we're ready to sync data, this makes it easy to swap over to the redux implmentations